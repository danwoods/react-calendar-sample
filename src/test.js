import React from 'react'
import Calendar from './index.js'
import renderer from 'react-test-renderer'

describe('Main calendar', () => {
  test('Basic Calendar without a provided date', () => {
    const component = renderer.create(<Calendar />)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
  test('Basic Calendar with a provided date', () => {
    const component = renderer.create(
      <Calendar date={new Date(2010, 10, 10)} />
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
