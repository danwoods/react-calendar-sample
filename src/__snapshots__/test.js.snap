// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main calendar Basic Calendar with a provided date 1`] = `
<div
  className="TEST-calendar-root "
>
  <div
    className="TEST-myh-root"
  >
    <h1>
      November 2010
    </h1>
  </div>
  <div
    className="TEST-dowlr-root"
  >
    <div
      className="TEST-dowlr-text"
    >
      S
    </div>
    <div
      className="TEST-dowlr-text"
    >
      M
    </div>
    <div
      className="TEST-dowlr-text"
    >
      T
    </div>
    <div
      className="TEST-dowlr-text"
    >
      W
    </div>
    <div
      className="TEST-dowlr-text"
    >
      T
    </div>
    <div
      className="TEST-dowlr-text"
    >
      F
    </div>
    <div
      className="TEST-dowlr-text"
    >
      S
    </div>
  </div>
  <div
    className="TEST-month-root"
  >
    <div
      className="TEST-week-root "
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="October 31, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            31
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 1, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            1
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 2, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            2
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 3, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            3
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 4, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            4
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 5, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            5
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 6, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            6
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="November 7, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            7
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 8, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            8
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 9, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            9
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 10, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            10
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 11, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            11
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 12, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            12
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 13, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            13
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="November 14, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            14
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 15, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            15
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 16, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            16
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 17, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            17
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 18, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            18
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 19, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            19
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 20, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            20
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="November 21, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            21
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 22, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            22
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 23, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            23
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 24, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            24
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 25, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            25
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 26, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            26
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 27, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            27
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="November 28, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            28
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 29, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            29
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="November 30, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            30
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="December 1, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            1
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="December 2, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            2
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="December 3, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            3
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="December 4, 2010"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            4
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Main calendar Basic Calendar without a provided date 1`] = `
<div
  className="TEST-calendar-root "
>
  <div
    className="TEST-myh-root"
  >
    <h1>
      March 2020
    </h1>
  </div>
  <div
    className="TEST-dowlr-root"
  >
    <div
      className="TEST-dowlr-text"
    >
      S
    </div>
    <div
      className="TEST-dowlr-text"
    >
      M
    </div>
    <div
      className="TEST-dowlr-text"
    >
      T
    </div>
    <div
      className="TEST-dowlr-text"
    >
      W
    </div>
    <div
      className="TEST-dowlr-text"
    >
      T
    </div>
    <div
      className="TEST-dowlr-text"
    >
      F
    </div>
    <div
      className="TEST-dowlr-text"
    >
      S
    </div>
  </div>
  <div
    className="TEST-month-root"
  >
    <div
      className="TEST-week-root "
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="March 1, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            1
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 2, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            2
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 3, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            3
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 4, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            4
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 5, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            5
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 6, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            6
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 7, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            7
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="March 8, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            8
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 9, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            9
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 10, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            10
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 11, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            11
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 12, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            12
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 13, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            13
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 14, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            14
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="March 15, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            15
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 16, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            16
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 17, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            17
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 18, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            18
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 19, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            19
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 20, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            20
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 21, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            21
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="March 22, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            22
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 23, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            23
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 24, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            24
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 25, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            25
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 26, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            26
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 27, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            27
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 28, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            28
          </h6>
        </div>
      </div>
    </div>
    <div
      className="TEST-week-root TEST-month-week-root"
      data-component-id="week"
    >
      <div
        className="TEST-day-root "
        title="March 29, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            29
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 30, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            30
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="March 31, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label "
          >
            31
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="April 1, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            1
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="April 2, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            2
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="April 3, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            3
          </h6>
        </div>
      </div>
      <div
        className="TEST-day-root TEST-week-day-root"
        title="April 4, 2020"
      >
        <div
          className="TEST-day-content "
        >
          <h6
            className="TEST-day-date-label TEST-different-month"
          >
            4
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
